src/openforms/submissions/admin.py

```diff
@@ -361,7 +361,7 @@ def successfully_processed(self, obj) -> bool | None:
 
     @admin.display(description=_("Registration backend"))
     def get_registration_backend(self, obj):
-        return obj.form.registration_backend or "-"
+        return backend if (backend := obj.form.registration_backends.first()) else "-"
```

Comment:
looks like this was bugged in the old code, it should really use `submission.registration_backend` so it takes into account the dynamic backend selection from logic rules (!)

Reply:
Check, the `obj.registration_backend` is now used for this admin display. I've also added a test that makes sure that the admin correctly shows the registration_backend of the submission